### PR TITLE
DAOS-9221 test: remove self._env_names (#10634)

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: pr daosracer online_extend online_parallel online_reintegration test_daos_run_io_conf test_unaligned_io test_daos_run_io_conf daosperf dfuse,daily_regression ior,daily_regression macsio mdtest,daily_regression Skip-unit-tests: true
Skip-fault-injection-test: true

Replace self._env_names and supporting functions by directly updating D_LOG_FILE as needed.

Signed-off-by: Dalton Bohning <dalton.bohning@intel.com>

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficent testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/util/daos_perf_base.py [hunk 1]
```diff
@@ -36,7 +37,8 @@ def run_daos_perf(self):
         daos_perf.cont_uuid.update(self.container.uuid)
         daos_perf.dmg_config_file.update(dmg_config_file)
         self.log.info("daos_perf command: %s", str(daos_perf))
-        daos_perf_env = daos_perf.get_environment(self.server_managers[0])
+        daos_perf_env = daos_perf.env.copy()
+        daos_perf_env["D_LOG_FILE"] = get_log_file("{}_daos.log".format(daos_perf.command))
```
Comment (phender) [inline src/tests/ftest/util/daos_perf_base.py:41]: I realize this is a cherry-pick, so I don't think we should change this now, but given the scope of the `daos_perf` object shouldn't his just be:
```suggestion
        daos_perf.env["D_LOG_FILE"] = get_log_file("{}_daos.log".format(daos_perf.command))
```
and then later:
```
        self.job_manager.assign_environment(daos_perf.env)
```

**Review by phender — APPROVED** [review]
looks good

File: src/tests/ftest/util/daos_perf_base.py (hunk 1 above)
Comment (daltonbohning) [reply src/tests/ftest/util/daos_perf_base.py:41]: You're right. I didn't notice that :)

**Review by shimizukko — APPROVED** [review]

**Comment by daltonbohning** [comment]
Latest test results: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10828/2/testReport/
With tags: 
```
pr daosracer online_extend online_parallel online_reintegration test_daos_run_io_conf
test_unaligned_io test_daos_run_io_conf daosperf dfuse,daily_regression ior,daily_regression
macsio mdtest,daily_regression soak_smoke
```
Passed, except for OSA tests, skipped due to:
- OSAOnlineDrain - https://daosio.atlassian.net/browse/DAOS-7289
- OSAOnlineExtend - https://daosio.atlassian.net/browse/DAOS-7195
- OSAOnlineParallelTest - https://daosio.atlassian.net/browse/DAOS-6664
- OSAOnlineReintegration - https://daosio.atlassian.net/browse/DAOS-7420

Which I think is safe, since the changes are similar to other test files.

**Pull request merged**